cli/src/node/mod.rs

```diff
@@ -434,10 +434,10 @@ impl Node {
             .with_provider(
                 collator
                     .new_sync_point(CollatorSyncContext::Historical)
-                    .chain(archive_block_provider)
+                    .chain(archive_block_provider.clone())
                     .chain(collator.new_sync_point(CollatorSyncContext::Recent))
                     .chain((
-                        blockchain_block_provider,
+                        blockchain_block_provider.cycle(archive_block_provider),
```

Comment:
`ArchiveBlockProvider` must reset its own state before it is going to return `None` on `get_next_block`